march_joint_inertia_controller/include/march_joint_inertia_controller/inertia_estimator.h

```diff
@@ -42,83 +42,83 @@ class InertiaEstimator
-  void setAcc_size(size_t acc_size);
-  void setNodeHandle(ros::NodeHandle& nh);
-  /**
-   * \brief Initialize the publisher with a name
-   */
-  void configurePublisher(std::string name);
-  void publishInertia();
+  void setAccSize(size_t acc_size);
 
   /**
    * \brief Applies the Butterworth filter over the last two samples and returns the resulting filtered value
    */
-  void apply_butter();
+  void applyButter();
 
   /**
    * \brief Estimate the inertia using the acceleration and torque
    */
-  void inertia_estimate();
+  void inertiaEstimate();
   /**
    * \brief Calculate a discrete derivative of the speed measurements
    */
-  void discrete_speed_derivative(double velocity, const ros::Duration& period);
+  double discreteSpeedDerivative(const ros::Duration& period);
   /**
    * \brief Calculate the alpha coefficient for the inertia estimate
    */
-  double alpha_calculation();
+  double alphaCalculation();
   /**
    * \brief Calculate the inertia gain for the inertia estimate
    */
-  double gain_calculation();
+  double gainCalculation();
   /**
    * \brief Calculate the correlation coefficient of the acceleration buffer
    */
-  void correlation_calculation();
+  void correlationCalculation();
   /**
    * \brief Calculate the vibration based on the acceleration
    */
-  double vibration_calculation();
+  double vibrationCalculation();
 
   /**
    * \brief Fill the rolling buffers with corresponding values.
    */
-  bool fill_buffers(double velocity, double effort, const ros::Duration& period);
+  void fillBuffers(double velocity, double effort, const ros::Duration& period);
 
   /**
    * \brief Calculate the initial correlation coefficient
    */
-  void init_p(unsigned int samples);
+  void initP(unsigned int samples);
 
   // Vector to be filled with samples of acceleration to determine the standard deviation from
   std::vector<double> standard_deviation;
 
 private:
-  ros::NodeHandle nh_;
-  ros::Publisher pub_;
-
-  float min_alpha_ = 0.4;  // You might want to be able to adjust this value from a yaml/launch file
-  float max_alpha_ = 0.9;  // You might want to be able to adjust this value from a yaml/launch file
+  double min_alpha_ = 0.4;  // You might want to be able to adjust this value from a yaml/launch file
+  double max_alpha_ = 0.9;  // You might want to be able to adjust this value from a yaml/launch file
 
   // This is a sixth order butterworth filter with a cutoff frequency at 15Hz in Second Order Sections form
-  double sos_[3][6] = {
-    { 2.31330497e-05, 4.62660994e-05, 2.31330497e-05, 1.00000000e+00, -1.37177561e+00, 4.75382129e-01 },
-    { 1.00000000e+00, 2.00000000e+00, 1.00000000e+00, 1.00000000e+00, -1.47548044e+00, 5.86919508e-01 },
-    { 1.00000000e+00, 2.00000000e+00, 1.00000000e+00, 1.00000000e+00, -1.69779140e+00, 8.26021017e-01 }
+  std::vector<std::vector<double>> sos_{
+    { 9.16782507e-09, 1.83356501e-08, 9.16782507e-09, 1.00000000e+00, -1.82520938e+00, 8.33345838e-01 },
```

Comment:
Why have these numbers changed, is it because of tuning? If so they should be put in a config file I think.

Reply:
These numbers changed due to the update frequency with which we acquire them. It produces a different filter when the input frequency is different.